### DISCUSSION TITLE
Adding Cosmos DB later

### DISCUSSION THREAD

**Discussion by ThoSt81** [microsoft/sample-app-aoai-chatGPT#1077] [Q&A]
Hi all,

for some reason I fail to deploy a new webapp and since everything else is working I wanted to simply add the Cosmos DB to enable the chat history for my environment.
Does anyone have a step-by-step-guide how to set this up for an existing webapp?

Thanks!

**Comment by dmytroHexagon** [comment]
I believe the deployment problem might have been solved with https://github.com/microsoft/sample-app-aoai-chatGPT/commit/d2ff7ad92b2719bac731d062f50718c647140c3e

If you already have a deployed solution, you can try to create a cosmosdb by yourself.

You will want to create corresponding database and collection in cosmos db:
![image](https://github.com/user-attachments/assets/dcc166e7-8f4f-4091-afdb-c379764f1544)

In the existing web app you will want to set next environment variables:
![image](https://github.com/user-attachments/assets/069075f5-bea1-4522-859a-2318c1e2b6e6)

You can find the key here in the cosmos db here:
![image](https://github.com/user-attachments/assets/2ea927f8-15a0-4748-a7cc-8be6ef19ba55)

Also please ensure you web app has system assigned identity "on"
![image](https://github.com/user-attachments/assets/7164190d-ca0b-40ca-bb97-36681d995b66)
